dynawo/sources/Modeler/DataInterface/IIDM/DYNVscConverterInterfaceIIDM.cpp

```diff
@@ -119,6 +128,34 @@ VscConverterInterfaceIIDM::getQ() {
   return InjectorInterfaceIIDM<IIDM::VscConverterStation>::getQ();
 }
 
+double
+VscConverterInterfaceIIDM::getQMax() {
+  if (vscConverterIIDM_.has_minMaxReactiveLimits()) {
+    return vscConverterIIDM_.minMaxReactiveLimits().max();
+  } else if (vscConverterIIDM_.has_reactiveCapabilityCurve()) {
+    assert(vscConverterIIDM_.reactiveCapabilityCurve().size() > 0);
+    double qMax = 0;
+    const double pGen = - getP();
+    const IIDM::ReactiveCapabilityCurve& reactiveCurve = vscConverterIIDM_.reactiveCapabilityCurve();
+    if (pGen <= reactiveCurve[0].p) {
+      qMax = reactiveCurve[0].qmax;
+    } else if (pGen > reactiveCurve[reactiveCurve.size() - 1].p) {
+      qMax = reactiveCurve[reactiveCurve.size() - 1].qmax;
+    } else {
+      for (unsigned int i = 0; i <= reactiveCurve.size() - 2; ++i) {
+        IIDM::ReactiveCapabilityCurve::point current_point = reactiveCurve[i];
+        IIDM::ReactiveCapabilityCurve::point next_point = reactiveCurve[i+1];
+        if (current_point.p <= pGen && next_point.p >= pGen) {
+          qMax = current_point.qmax + (pGen - current_point.p) * (next_point.qmax - current_point.qmax) / (next_point.p - current_point.p);
+        }
+      }
+    }
+    return qMax;
+  } else {
```

Comment:
useless else

Reply:
Shouldn't we still give a value for qMax when the vsc has no reactivelimits and no reactivecapabilitycurve just like for the generator ?

Reply:
I mean to only remove the else not the pMax assignement

Reply:
But in that case, if has_minMaxReactiveLimits is false and has_reactiveCapabilityCurve is false too we don't return anything no ?

Reply:
I mean replace "} else { return 0;}" by "return 0;" which is an equivalent code

Reply:
Ok thanks